tests/constructor/test_load_checkpoint.py

```diff
@@ -0,0 +1,251 @@
+import unittest
+
+import torch
+import copy
+from typing import *
+from collections import OrderedDict
+
+from .checkpoints.model import Model
+from src.constructor.load import load_checkpoint, generate_require_state_dict
+
+
+def compare_state_dicts(current_state_dict: Dict[str, torch.Tensor], load_state_dict: Dict[str, torch.Tensor], 
+                        check_keys: Optional[List[str]] = None):
+    """Compare state dicts by check_keys.
+
+    Assert if check keys is not equal.
+
+    Args:
+        current_state_dict: First compared state dict.
+        load_state_dict: Second compared state dict.
+        check_keys: The keys for compare.
+    """
+    check_keys = list(current_state_dict.keys()) if check_keys is None else check_keys
+    for key in check_keys:
+        assert torch.equal(current_state_dict[key], load_state_dict[key])
+
+
+def load_state_dict_with_prefix(state_dict_path: str, prefix: str) -> Dict[str, torch.Tensor]:
+    """Load state dict with prefix for keys.
+    Args:
+        state_dict_path: Path for loaded state dict.
+        prefix: Prefix for keys.
+
+    Returns:
+        prefix_state_dict: Loaded state dict from state_dict_path with prefix in dictionary keys.
+    """
+    state_dict = torch.load(state_dict_path)
+    prefix_state_dict = OrderedDict()
+    for key, value in state_dict.items():
+        prefix_state_dict[prefix + '.' + key] = value
+    return prefix_state_dict
+
+
+class TestCheckpoint(unittest.TestCase):
+    base_path = 'tests/constructor/checkpoints/base.pth'
+    layer_path = 'tests/constructor/checkpoints/layer.pth'
+
+    model_keys = [
+        'layer1.module.conv1.weight', 
+        'layer1.linear.weight',
+        'linear.weight',
+    ]
+    
+
+    def test_generate_require_state_dict_when_only_base_checkpoint_was_defined(self):
+        base_state_dict = {
```

Comment:
DRY. In several test function i saw base_state_dict with identity keys and values. Move to class property( like model_keys)